### PR TITLE
Add clueweb task

### PR DESCRIPTION
**Pull request by ZanezZephyrs**
Remove the mc4 aux files. Now the tasks list the files in a bucket dir. 
Added the "clueweb_small" task, i.e, the task with 2% of the clueweb that we currently have.

### CODE REVIEW

File: t5x/tasks.py [hunk 1]
```diff
@@ -2,12 +2,21 @@
...
-# from t5x import custom_tasks
+# from t5x import tasks
 
+def list_files_in_gcs_dir(bucket, prefix):
+    # From https://github.com/GoogleCloudPlatform/google-cloud-python/issues/920
+    client = storage.Client()
+    iterator = client.list_blobs(bucket, prefix=prefix, delimiter="/")
+    paths=[]
+    for i in iterator:
+        full_path=f"gs://{bucket}/{i.name}"
```
Comment (rodrigonogueira4) [inline t5x/tasks.py:17]: format with lint?

Comment (rodrigonogueira4) [reply t5x/tasks.py:17]: i.e., whitespace between `=`

File: t5x/tasks.py [hunk 2]
```diff
@@ -18,55 +27,63 @@ def cast(x, features=["targets"],dtype=tf.int32):
...
+        preprocessors=[
+            functools.partial(
+                seqio.preprocessors.rekey, key_map={
+                    "inputs": None,
+                    "targets": "text"
+                }),
+            # seqio.preprocessors.tokenize, # to tokenize on the fly
+            functools.partial(
+                cast, features=["targets"]), # cast function is necessary to adjust tf.dtype when loading pre-tokenized data
+            # seqio.CacheDatasetPlaceholder(), # not yet sure of the effects
```
Comment (rodrigonogueira4) [inline t5x/tasks.py:77]: remove?

File: t5x/tasks.py [hunk 3]
```diff
@@ -18,55 +27,63 @@ def cast(x, features=["targets"],dtype=tf.int32):
...
+    task_split_to_filepattern={
+        "train": train_paths
+    }
+
+    # all paths for validation data
+    if task_parameters.get("val_prefix",None):
+        val_paths=list_files_in_gcs_dir(task_parameters["bucket"], task_parameters["val_prefix"])
+        task_split_to_filepattern["validation"]=val_paths
+
+    # add pt_mc4 task
```
Comment (rodrigonogueira4) [inline t5x/tasks.py:60]: Capitalize comments and add period at the end?

**Comment by rodrigonogueira4** [comment]
@ZanezZephyrs thanks for submitting this!
Just minor comments.. I think we should start using linting/pep8 to avoid minor comments in the future

File: t5x/tasks.py [hunk 4]
```diff
@@ -18,55 +27,63 @@ def cast(x, features=["targets"],dtype=tf.int32):
...
+task_data={
+    "pt_mc4":{
+        "bucket": "gpt-j-pt",
+        "train_prefix":"data/tfrecords-mc4-total-gptimbau/train/mc4-gptimbau_",
+        "val_prefix": "data/tfrecords-mc4-total-gptimbau/val/mc4-gptimbau_",
+    },
+    "clueweb_small": {
+        "bucket": "maritica-europe-west-4",
+        "train_prefix":"datasets/tokenized/clueweb_small/train/clueweb_",
+        "val_prefix": None,
```
Comment (rodrigonogueira4) [inline t5x/tasks.py:40]: I think we need a small validation set.. maybe the first file?

Comment (ZanezZephyrs) [reply t5x/tasks.py]: Added, i am using only the first tfrecord for validation.

**Comment by ZanezZephyrs** [comment]
> @ZanezZephyrs thanks for submitting this! Just minor comments.. I think we should start using linting/pep8 to avoid minor comments in the future

@rodrigonogueira4  i believe the formatting problems should be fixed now.
Currently i am formating the files with black on demand, i could set up the pre-commit in the repository to automatically run on commit.

**Comment by rodrigonogueira4** [comment]
> @rodrigonogueira4 i believe the formatting problems should be fixed now. Currently i am formating the files with black on demand, i could set up the pre-commit in the repository to automatically run on commit.

That would be great!

File: t5x/tasks.py [hunk 5]
```diff
@@ -2,71 +2,89 @@
 import functools
 import tensorflow as tf
 from seqio import utils
-
+from google.cloud import storage
 
 # IN ORDER TO USE TASKS DEFINED HERE
 # YOU NEED TO IMPORT THIS FILE IN THE GIN CONFIGURATION
-# from t5x import custom_tasks
+# from t5x import tasks
```
Comment (hugoabonizio) [inline t5x/tasks.py:9]: Is this a leftover comment?

Comment (ZanezZephyrs) [reply t5x/tasks.py:9]: it is meant to be an example of how you should import the tasks in the gin file rsrs.

Comment (rodrigonogueira4) [reply t5x/tasks.py:9]: I think we can remove as it may cause some confusion?

Comment (ZanezZephyrs) [reply t5x/tasks.py]: Okay, removed.

**Review by hugoabonizio — APPROVED** [review]

**Pull request merged**